src/abstractdataframe/show.jl

```diff
@@ -410,21 +404,27 @@ function showrows(io::IO,
     colwidths = maxwidths .+ textwidth(" ") .+ textwidth(" |")  # padding+border
     colwidths[end] += textwidth("|")  # add row column initial border
     coltextwidths = colwidths[end] .+ colwidths[1:end-1]
-    coloverflows = coltextwidths .> displaysize(io)[2]
-    chunkbounds = getchunkbounds(maxwidths, splitcols || !allcols, displaysize(io)[2])
 
-    nchunks = length(chunkbounds) - 1
-    nchunks = allcols ? nchunks : (chunkbounds[2] < 1 ? 0 : min(nchunks, 1))
-    nchunks = !allcols && splitcols && any(coloverflows) ? 0 : nchunks
+    chunkedcols = getchunkedcols(maxwidths, splitcols || !allcols, displaysize(io)[2])
+    chunkwidths = [colwidths[end] + sum(colwidths[cols]) for cols=chunkedcols]
+    chunkoverflows = chunkwidths .> displaysize(io)[2]
+
+    if !allcols && any(chunkoverflows)
+        chunkedcols = chunkedcols[1:(findfirst(chunkoverflows) - 1)]
+    end
+
+    if !allcols && length(chunkedcols) > 0
+        chunkedcols = chunkedcols[[1]]
+    end
 
     header = displaysummary ? summary(df) : ""
-    nomit = ncols - sum(chunkbounds[1:nchunks+1])
+    nomit = length(chunkedcols) > 0 ? ncols - sum(length.(chunkedcols)) : ncols
```

Comment:
maybe:
```suggestion
    nomit = isempty(chunkedcols) ? ncols : ncols - sum(length, chunkedcols)
```
it seems cleaner and should allocate less